From 76d318d7748bbbb86123d5baa0514b7e7aeac94c Mon Sep 17 00:00:00 2001
From: tpaviot <tpaviot@gmail.com>
Date: Sun, 3 Jun 2012 17:14:25 +0200
Subject: [PATCH] [clang-warning-fix][constant-logical-operand]

Use of logical '||' with constant operand. Fix: use '|' for a bitwise
operation.
---
 src/TopOpeBRepTool/TopOpeBRepTool_makeTransition.cxx |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

Index: oce/src/TopOpeBRepTool/TopOpeBRepTool_makeTransition.cxx
===================================================================
--- oce.orig/src/TopOpeBRepTool/TopOpeBRepTool_makeTransition.cxx
+++ oce/src/TopOpeBRepTool/TopOpeBRepTool_makeTransition.cxx
@@ -270,14 +270,14 @@ static Standard_Boolean FUN_mkT2dquad(co
   // = sta = state of point on e1 after pt / e2
 
   gp_Dir tga1,tga2;
-  Standard_Boolean mk1 = isINifh1 || isON2ifss || isIN2ifss;
+  Standard_Boolean mk1 = isINifh1 | isON2ifss | isIN2ifss;
   if (mk1) {
     Standard_Integer st1 = 0; gp_Dir tgnear1;
     Standard_Boolean ok = FUN_tg(e1,par1,f1,l1,factor,tgnear1,st1);
     if (!ok) return Standard_False;
     tga1 = (st1 == AFTER) ? tgnear1 : tgnear1.Reversed();    
   }
-  Standard_Boolean mk2 = isINifh2 || isON2ifss || isOU2ifss;
+  Standard_Boolean mk2 = isINifh2 | isON2ifss | isOU2ifss;
   if (mk2) {
     Standard_Real f2,l2; FUN_tool_bounds(e2,f2,l2);
     Standard_Integer st2 = 0; gp_Dir tgnear2;
